### PR TITLE
FEAT-#4359: Add __dataframe__ method to the protocol dataframe

### LINKED ISSUE #4359: Add __dataframe__ method to the protocol dataframe
```python
import modin.pandas as pd
from modin.pandas.utils import from_dataframe
df = pd.DataFrame({"a": [1, 2, 3]})
dfx = df.__dataframe__()
from_dataframe(dfx)
ValueError: `df` does not support DataFrame exchange protocol, i.e. `__dataframe__` method
```

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4359  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by dchigarev — COMMENTED** [review]
Looks fine, left a single comment

File: modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py [hunk 1]
```diff
@@ -376,6 +376,32 @@ class ProtocolDataframe(ABC):
...
+            to make a copy of the data. For example, copying data would be necessary
+            if a library supports strided buffers, given that this protocol
+            specifies contiguous buffers. Currently, if the flag is set to ``False``
+            and a copy is needed, a ``RuntimeError`` will be raised.
+
+        Returns
+        -------
+        ProtocolDataframe
+        """
+
```
[the remark this answers is not in the document]
Comment (YarShev) [reply modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py:404]: sure, forgot to add 😄

File: modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py [hunk 2]
```diff
@@ -376,6 +376,32 @@ class ProtocolDataframe(ABC):
...
+            to make a copy of the data. For example, copying data would be necessary
+            if a library supports strided buffers, given that this protocol
+            specifies contiguous buffers. Currently, if the flag is set to ``False``
+            and a copy is needed, a ``RuntimeError`` will be raised.
+
+        Returns
+        -------
+        ProtocolDataframe
+        """
+        pass
```
Comment (YarShev) [inline modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py:404]:
```suggestion
        pass

```

**Review by dchigarev — APPROVED** [review]
LGTM, will merge on green CI

File: modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py (hunk 1 above)
Comment (dchigarev) [inline modin/core/dataframe/base/exchange/dataframe_protocol/dataframe.py:404]:
```suggestion
        """
        pass

```

**Review by dchigarev — APPROVED** [review]

**Pull request merged**